lib/shift/circuit_breaker/circuit_handler.rb

```diff
@@ -13,34 +13,39 @@ module CircuitBreaker
     # the defined skip_duration, ie. no operations are executed and the provided fallback is called.
     #
     class CircuitHandler
-      attr_accessor :name, :error_threshold, :skip_duration, :exception_classes, :error_count, :last_error_time, :state, :logger, :monitor
+      attr_accessor :name, :error_threshold, :skip_duration, :exception_classes, :enable_error_logging, :error_count
+      attr_accessor :last_error_time, :state, :logger, :monitor
 
       DEFAULT_EXCEPTION_CLASSES = [Net::OpenTimeout, Net::ReadTimeout, Faraday::TimeoutError, Timeout::Error].freeze
+      DEFAULT_ERROR_LOGGING_STATE = true.freeze
 
       # Initializer creates an instance of the service
       #
       # @param [Symbol]  name              - the name used to identify the circuit breaker
       # @param [Integer] error_threshold   - The minimum error threshold required for the circuit to be opened/tripped
       # @param [Integer] skip_duration     - The duration in seconds the circuit should be open for before operations are allowed through/executed
       # @param [Array]   additional_exception_classes - Any additional exception classes to rescue along the DEFAULT_EXCEPTION_CLASSES
+      # @param [Boolean] enable_error_logging         - Decided whether to log errors or not. Still they will be monitored
       # @param [Object]  logger            - service to handle error logging
       # @param [Object]  monitor           - service to monitor metric
       def initialize(name,
                      error_threshold:,
                      skip_duration:,
                      additional_exception_classes: [],
+                     enable_error_logging: DEFAULT_ERROR_LOGGING_STATE,
                      logger: Shift::CircuitBreaker::CircuitLogger.new,
                      monitor: Shift::CircuitBreaker::CircuitMonitor.new)
 
-        self.name               = name
-        self.error_threshold    = error_threshold
-        self.skip_duration      = skip_duration
-        self.exception_classes  = (additional_exception_classes | DEFAULT_EXCEPTION_CLASSES)
-        self.logger             = logger
-        self.monitor            = monitor
-        self.error_count        = 0
-        self.last_error_time    = nil
-        self.state              = :closed
+        self.name                 = name
+        self.error_threshold      = error_threshold
+        self.skip_duration        = skip_duration
+        self.enable_error_logging = enable_error_logging
```

Comment:
Also what's the benefit of doing this and using an attr accessor rather than just using the constant directly on line 112?

Reply:
Here the default value is true, but if we want to disable it, we will send it as false via initializer.